sdks/go/pkg/beam/util/starcgenx/starcgenx.go

```diff
@@ -736,7 +736,7 @@ func (e *Extractor) makeInput(sig *types.Signature) (shimx.Input, bool) {
 // deref returns the string identifier for the element type of a pointer var.
 // deref panics if the var type is not a pointer.
 func (e *Extractor) deref(v *types.Var) string {
-	p := v.Type().(*types.Pointer)
+	p := types.Unalias(v.Type()).(*types.Pointer)
 	return types.TypeString(p.Elem(), e.qualifier)
 }
```

Comment:
I think this change is unnecessary and will break things since it's already expecting a pointer to a value and is trying to get the name of the pointer type anyway. 

Given types like this:

```
type A = someType

var a *A
```

should still produce "A" for this case.